src/devices/llama_devices.ts

```diff
@@ -0,0 +1,165 @@
+        ],
+    });
+}
+
+function mapModernEnum(cfg: Config, epName: string) {
+    return modernExtend.enumLookup({
+        name: cfg.name,
+        lookup: mapObject(cfg.values, (k) => k.name, (v) => v.value),
+        endpointName: epName,
+        cluster: cfg.cluster,
+        attribute: cfg.attribute,
+        description: cfg.description,
+        access: 'ALL',
+        entityCategory: 'config',
+    });
+}
+
+function addEndpointPrefix(ep: number) {
+    return `${prefixEnpointName}${ep}`;
+}
+
+function mapObject<T>(arr: T[], mapKey: (k: T) => string | [string], mapValue: (v: T) => string | number) {
+    return Object.fromEntries(arr.map((el, i, arr) => [mapKey(el), mapValue(el)]));
+}
+
+const fzLocal = {
+    on_off_action: {
+        cluster: 'genOnOff',
+        type: [
+            'commandOn',
+            'commandOff',
+            'commandToggle',
+            // 'commandOnWithTimedOff'
+        ],
+        convert: (model, msg, publish, options, meta) => {
+            return {action: `${msg.endpoint.ID}_${msg.type.replace('command', '')}`};
+        },
+    } satisfies Fz.Converter,
+};
+
+const definitions: Definition[] = [
+    {
+        zigbeeModel: ['Zucchini'],
+        model: 'Zucchini',
+        vendor: 'Llama Devices',
+        description: 'On/Off output device',
+        fromZigbee: [fzLocal.on_off_action],
+        exposes: [
+            e.action([]),
+        ],
+        extend: [
+            mapModernActionEnum(),
+            modernExtend.battery(),
+            modernExtend.temperature(),
+            modernExtend.humidity(),
+            ...((epCount: number) => {
+                const features: ModernExtend[] = [];
+                for (let ep = 1; ep <= epCount; ep++) {
+                    for (let i = 0; i < configs.length; i++) {
+                        features.push(mapModernEnum(configs[i], addEndpointPrefix(ep)));
+                    }
+                }
+                return features;
+            })(endpointCount),
+        ],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            const ep1 = device.getEndpoint(1);
+            await reporting.bind(ep1, coordinatorEndpoint, ['genPowerCfg']);
+            await reporting.bind(ep1, coordinatorEndpoint, ['msRelativeHumidity']);
+            await reporting.bind(ep1, coordinatorEndpoint, ['msTemperatureMeasurement']);
+            device.endpoints.filter((ep) => ep.ID != 242)
+                .forEach(async (ep) => {
+                    await ep.bind('genOnOff', coordinatorEndpoint);
+                    await ep.read('genOnOffSwitchCfg', configs.map((c) => c.attribute));
+                });
+            await reporting.batteryPercentageRemaining(ep1, {min: 0, max: 3600, change: 0});
+            await reporting.temperature(ep1, {min: 0, max: 3600, change: 0});
+            await reporting.humidity(ep1, {min: 0, max: 3600, change: 0});
+            device.save();
+        },
+        endpoint: (device) => {
```

Comment:
There is already `deviceEndpoints` Modern Extend for this.

Reply:
yes, you are right. But, I don't want min 10 seconds, I want 0 seconds for min. Battery handles it, but other hardcoded `min: '10_SECONDS'`

Reply:
but `deviceEndpoints` doesn't provide `device` instance. I don't want hardcode. I already hardcoded `enpointCount` and it hearts... :(

Reply:
Ah, I see, oh well. Will be refactored later I suppose